docs/sql-data-sources-protobuf.md

```diff
@@ -0,0 +1,301 @@
+---
```

Comment:
Could you check the formatting. It seems to require fixes. 
Sample from github view of this page below. It it not rendered correctly.
<img width="1410" alt="image" src="https://user-images.githubusercontent.com/502522/207749158-8990d41c-4d60-411d-913f-232b432df2de.png">

Reply:
@rangadi it appears to be the same for spark-avro as well, https://github.com/apache/spark/blob/master/docs/sql-data-sources-avro.md#to_avro-and-from_avro. It will be rendered as a webpage rather than a readme file. spark-avro documentation can be found at https://spark.apache.org/docs/latest/sql-data-sources-avro.html. not sure how to test

Reply:
https://github.com/apache/spark/blob/master/docs/README.md
This doc guides through how you can create the final web pages from MD files.

Reply:
@HeartSaVioR thanks for the doc link, I have made changes and verified the generated docs/_site/sql-data-sources.html  and ddocs/_site/sql-data-sources-protobuf.html pages, both look good. 

cc: @rangadi

Reply:
This is great! How do I see generate these html pages myself?

Reply:
@rangadi 
Mac OS
```
sudo gem install bundler
cd spark/docs
bundle install
SKIP_API=1 bundle exec jekyll build     
open -a "Google Chrome" _site/sql-data-sources.html
```

Reply:
That worked. 
Thanks for adding the documentation. It is really important. I will leave a few comments.